### ISSUE TITLE
Improve profile switching when multiple profiles have a trigger with "switch back" enabled

### ISSUE THREAD

**Issue by matlo** [matlo/GIMX#332]
Multiple "switch back" triggers don't work well, because gimx remembers a single previous config.

I'll consider the following improvement in a future release:
* remember the previous & next profile numbers for each profile (-1 = no profile)
* when an event matches the switch back event of profile X:
 * if X is active, switch back to previous(X)
 * if X is not active, update previous(next(X)) to previous(X)

In other words, have an history of profile switches, update that history when switch back events happen, and only switch back if the switch back event matches the one of the active profile.
